src/Storages/StorageDistributed.cpp

```diff
@@ -893,28 +894,6 @@ void StorageDistributed::read(
             storage_snapshot,
             processed_stage);
 
```

Comment:
Maybe put this code into a function like `getShardFilterGeneratorForCustomKey()` and use it where needed. Then it'd not be necessary to put custom specific code into `ClusterProxe::executeQuery()`